pkg/account/delete/delete.go

```diff
@@ -60,55 +67,121 @@ func (a Account) String() string {
 	return fmt.Sprintf("%q (%s)", a.Name, a.UUID)
 }
 
-// AccountResources removes all given Resources from the given Account
+// DeleteAccountResources removes all given Resources from the given Account
 // Returns an error if any resource fails to be deleted, but attempts to delete as many resources as possible and only returns an error at the end.
-func AccountResources(ctx context.Context, account Account, resourcesToDelete Resources) error {
+func DeleteAccountResources(ctx context.Context, account Account, resourcesToDelete Resources) error {
+	totalErrorCount := deleteUsers(ctx, account, resourcesToDelete.Users) +
+		deleteServiceUsers(ctx, account, resourcesToDelete.ServiceUsers) +
+		deleteGroups(ctx, account, resourcesToDelete.Groups) +
+		deleteAccountPolicies(ctx, account, resourcesToDelete.AccountPolicies) +
+		deleteEnvironmentPolicies(ctx, account, resourcesToDelete.EnvironmentPolicies)
 
-	deleteErrors := 0
+	if totalErrorCount > 0 {
+		return fmt.Errorf("encountered %d errors - please check logs for details", totalErrorCount)
+	}
+	return nil
+}
 
-	for _, user := range resourcesToDelete.Users {
-		if err := account.APIClient.DeleteUser(ctx, user.Email.Value()); err != nil && errors.Is(err, NotFoundErr) {
-			log.Info("User %q does not exist for account %s", user.Email, account)
-		} else if err != nil {
-			log.Error("Failed to delete user %q from account %s: %v", user.Email, account, err)
-			deleteErrors++
-		} else {
+func deleteUsers(ctx context.Context, account Account, users []User) int {
+	errCount := 0
+	for _, user := range users {
+		err := account.APIClient.DeleteUser(ctx, user.Email.Value())
+		if err == nil {
 			log.Info("Deleted user %q from account %s", user.Email, account)
+			continue
 		}
+
+		notFoundErr := &ResourceNotFoundError{}
+		if errors.As(err, &notFoundErr) {
+			log.Info("User %q does not exist for account %s", user.Email, account)
+			continue
+		}
+
+		log.Error("Failed to delete user %q from account %s: %v", user.Email, account, err)
+		errCount++
 	}
-	for _, group := range resourcesToDelete.Groups {
-		if err := account.APIClient.DeleteGroup(ctx, group.Name); err != nil && errors.Is(err, NotFoundErr) {
-			log.Info("Group %q does not exist for account %s", group.Name, account)
-		} else if err != nil {
-			log.Error("Failed to delete group %q from account %s: %v", group.Name, account, err)
-			deleteErrors++
-		} else {
+	return errCount
+}
+
+func deleteServiceUsers(ctx context.Context, account Account, serviceUsers []ServiceUser) int {
+	errCount := 0
+	for _, user := range serviceUsers {
+		err := account.APIClient.DeleteServiceUser(ctx, user.Name)
```

Comment:
Should we maybe refactor this one a bit? Currently, it fetches all available service users for each entry in the service users that should be deleted

Reply:
I have added this to our follow ups 👍